Cargo.toml

```diff
@@ -17,18 +18,21 @@ maintenance = { status = "passively-maintained" }
 
 [features]
 default = ["tokio"]
-tokio = ["futures", "libc", "mio", "tokio-core"]
+tokio = ["futures", "libc", "dep:tokio"]
 
 [build-dependencies]
 cc = "~1"
 
 [dependencies]
-futures = { version = "~0.1", optional = true }
+futures = { version = "~0.3", default-features = false, optional = true }
 libc = { version = "~0.2", optional = true }
-mio = { version = "~0.6", optional = true }
-tokio-core = { version = "~0.1", optional = true }
+tokio = {version = "~1.29", features = ["io-util", "net"], optional = true }
```

Comment:
Is that `1.29` minimal version needed? What's the feature that is not in the lower ones?